kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessGenerator.java

```diff
@@ -268,105 +276,58 @@ private MethodCallExpr createProcessRuntime() {
-                }
-                // annotate for injection or add constructor for initialization
-                handler.findAll(FieldDeclaration.class).forEach(fd -> {
-                    if (useInjection()) {
-                        annotator.withInjection(fd);
-                    } else {
-                        BlockStmt constructorBody = new BlockStmt();
-                        AssignExpr assignExpr = new AssignExpr(
-                                                               new FieldAccessExpr(new ThisExpr(), fd.getVariable(0).getNameAsString()),
-                                                               new ObjectCreationExpr().setType(fd.getVariable(0).getType().toString()),
-                                                               AssignExpr.Operator.ASSIGN);
-                        
-                        constructorBody.addStatement(assignExpr);
-                        clazz.addConstructor(Keyword.PUBLIC).setBody(constructorBody);
-                    }
-                });
-                
-                
-                additionalClasses.add(handler);
-            });
-        }
+       
         if (!processMetaData.getGeneratedListeners().isEmpty()) {
+            BlockStmt body = new BlockStmt();
+            MethodDeclaration internalConfigure = new MethodDeclaration()
+                    .setModifiers(Modifier.Keyword.PUBLIC)
+                    .setType(targetTypeName)
+                    .setName("configure")
+                    .setBody(body);   
             
+            // always call super.configure
+            body.addStatement(new MethodCallExpr(new SuperExpr(), "configure"));
             processMetaData.getGeneratedListeners().forEach(listener -> {
-                
                 ClassOrInterfaceDeclaration clazz = listener.findFirst(ClassOrInterfaceDeclaration.class).get();
-                String packageName = listener.getPackageDeclaration().map(pd -> pd.getName().toString()).orElse("");
-                String clazzName = clazz.getName().toString();
-                
-                MethodCallExpr eventSupport = new MethodCallExpr(new NameExpr("services"), "getEventSupport");            
-                MethodCallExpr registerListener = new MethodCallExpr(eventSupport, "addEventListener").addArgument(new ObjectCreationExpr(null, new ClassOrInterfaceType(null, packageName + "." + clazzName), NodeList.nodeList()));
+                MethodCallExpr eventSupport = new MethodCallExpr(new NameExpr("services"), "getEventSupport");
+                MethodCallExpr registerListener = new MethodCallExpr(eventSupport, "addEventListener")
+                    .addArgument(
+                        new ObjectCreationExpr(
+                            null,
+                            new ClassOrInterfaceType(
+                                null,
+                                listener.getPackageDeclaration().map(pd -> pd.getName().toString()).orElse("") + "." +clazz.getName()),
```

Comment:
If `getPackageDeclaration()` returns an empty `Optional` this will produce an invalid string: `.ClassName`
Can you please verify if `getPackageDeclaration()` can be empty (i.e. default package)?
If not I think it is better to throw an exception like "This should never happen" instead of a empty string.

In general we can probably replace the code with this alternative
```suggestion
                                listener.getPackageDeclaration().map(pd -> pd.getName().toString() + "." + clazz.getName()).orElse(clazz.getName()),
```
Wdyt?